### PR TITLE
Issue #919 - Stop debug vm on end of debug session.

### PR DESCRIPTION
**Pull request by rorygraves**
Debug sessions launched by Ensime were not terminated when Ensime finishes debugging.
This commit makes the debugged vm stop when we terminate the debug vm session.
This may be a little agressive for attached sessions.

### CODE REVIEW

**Comment by aemoncannon** [comment]
Cool

**Comment by edani** [comment]
@rorygraves  I think for attached sessions, exit() shouldn't be called, just dispose().

**Comment by rorygraves** [comment]
@edani Yeah fair comment - will look at it when i get the chance.

**Comment by rorygraves** [comment]
lol, went to fix this over the weekend and found several other bugs - so bigger refactor in progress - closing for now.

File: core/src/main/scala/org/ensime/core/DebugManager.scala [hunk 1]
```diff
@@ -237,67 +239,75 @@ class DebugManager(
...
-      handleRPCWithVM() { vm =>
-        sender ! TrueResponse
+      sender ! handleRPCWithVM() { vm =>
+        TrueResponse
       }
-
     case DebugStopReq =>
-      handleRPCWithVM() { vm =>
+      sender ! handleRPCWithVM() { vm =>
+        if (vm.mode.shouldExit) {
```
Comment (rorygraves) [inline core/src/main/scala/org/ensime/core/DebugManager.scala]: @edani This fix means it should only trigger the exit on a VM started by Ensime - as per original discussion.

**Comment by rorygraves** [comment]
@fommil @edani @aemoncannon 
So this is a rework of the original fix - got a bit carried away, but DebugManager definately has less errors in it right now.  Fundamental change is that the ```sender !``` only occurs in the Receive block and everything else returns an RpcResponse and the type system helps to ensure this is the case.  I found a few cases where a non-response value or, in one case, no value would be returned.
So in summary, fixed the stop thing, cleaned up a bunch of other stuff...

**Comment by edani** [comment]
@rorygraves   This looks good to me. I'll see if I can reproduce the appveyor failure locally.

**Comment by fommil** [comment]
Should we stop all VM that we started in the post Stop ?

**Comment by rorygraves** [comment]
Excellent point. Yes.

**Comment by rorygraves** [comment]
Thanks @edani
